clang/include/clang-c/Dependencies.h

```diff
@@ -232,6 +230,19 @@ clang_experimental_DependencyScannerWorker_getFileDependencies_v1(
     const char *WorkingDirectory, CXModuleDiscoveredCallback *MDC,
     void *Context, CXString *error);
 
+/**
+ * Same as \c clang_experimental_DependencyScannerWorker_getFileDependencies_v1,
+ * but \c BuildArguments of \c CXFileDependencies contains the full Clang
+ * command line, not just additional arguments, and \c BuildArguments of each
+ * \c CXModuleDependency now contain the necessary '-fmodule-map-file='
+ * arguments.
+ */
+CINDEX_LINKAGE CXFileDependencies *
+clang_experimental_DependencyScannerWorker_getFileDependencies_v2(
+    CXDependencyScannerWorker Worker, int argc, const char *const *argv,
+    const char *WorkingDirectory, CXModuleDiscoveredCallback *MDC,
+    void *Context, CXString *error);
+
```

Comment:
Can you explain how the `-fmodule-map-file=` get filled in? It seems like something the build system should be in charge of, and it's not obvious to me whether `CXModuleDiscoveredCallback` gives the build system control of the names somehow.

Reply:
The `-fmodule-map-file=` arguments get filled internally by the dependency scanner now. These arguments are not necessary in the vast majority of cases. We only need them for modules that are marked as `[no_undeclared_includes]` (see the mentioned upstream patches). I know the intention was for the build system to generate these arguments, but it never did anything special: it simply filled in the reported top-level module maps for each of the dependencies.

What would be the use-case where this isn't the correct thing to do? Distributed builds?

Reply:
Heh; I was confused, and conflated `-fmodule-map-file` with `-fmodule-file`. I thought the build system was losing control of naming output artifacts (i.e., `PCM` files)... but that's not happening at all!